### DISCUSSION TITLE
`pyvista.read()` API change

### DISCUSSION THREAD

**Discussion by ahulist** [pyvista/pyvista#6886] [Q&A] [answered]
### First check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the PyVista documentation, with the integrated search.
- [X] I already searched in Google "How to X in PyVista" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I will NOT make duplicate discussions or issues in the future.

### Commit to Help

- [X] I commit to help with one of those options.

### What is the problem, question, or error?

Something in version 0.44.* must have changed somewhere, causing `pv.read().points` to return `pyvista_ndarray` instead of a VTK object (while loading an STL). I did not find any mention about it in release notes. Actually, I was under the impression that pyvista always returned an array, but please take a look to see the issue:
https://youtu.be/NYBiKcqMtt4

Object type returned by `pyvista.read().points` changed. I'm looking for an explanation for this (or correction where I made a mistake).

### System Information

```shell
--------------------------------------------------------------------------------
  Date: Wed Nov 27 18:30:29 2024 CET

                OS : Linux
            CPU(s) : 12
           Machine : x86_64
      Architecture : 64bit
       Environment : Python
       GPU Details : error
  MathText Support : True

  Python 3.9.18 (main, Sep 11 2023, 13:41:44)  [GCC 11.2.0]

           pyvista : 0.44.2
               vtk : 9.3.1
             numpy : 2.0.2
        matplotlib : 3.9.2
            scooby : 0.10.0
             pooch : 1.8.2
            pillow : 11.0.0
--------------------------------------------------------------------------------
```

**Comment by user27182** [comment, marked as answer]
`pyvista_ndarray` should always be returned. But, the recent release of VTK 9.4 has broken this in some cases, causing `points` to incorrectly return a VTK object. 

PyVista `0.44.2` has vtk pinned `<9.4.0` to address this since pyvista does not yet support vtk 9.4. So, that's why in your video the correct `pyvista_ndarray` output is returned with pyvista==0.44.2. But, if installing an older version (e.g. `0.43.10`), this will break pyvista since vtk is not pinned and 9.4 will be installed.

If installing any older version of pyvista (<0.44.2), then you must also pin `vtk<9.4.0`, e.g. install with
```
pip install pvista==0.43.10 vtk<9.4.0
```

**Reply by ahulist** [reply]
I thought I went crazy for a second. But that makes sense now, thanks for the explanation!
